mlflow/deployments/__init__.py

```diff
@@ -2,7 +2,7 @@
 from mlflow.deployments import interface
 
 
-# TODO: It's a good practise to avoid using ``list`` here. Looking for a better alternative
+# TODO: It's a good practise to avoid using ``list`` here
```

Comment:
Thanks for bringing this up, IMO we could go with `create_deployment`, `delete_deployment`, `update_deployment`, `list_deployments`, `describe_deployment` here. This matches what we do e.g. in the MlflowClient class / fluent API (`mlflow.list_experiments`, `MlflowClient.list_experiments`, etc). While some OSS libraries do seem to actually define methods named `list` (see e.g. the [docker Python client](https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.ImageCollection.list)), I agree that it'd be a bad idea (adding `from mlflow.deployments import *` would suddenly break all usages of lists in your program).

Alternatively, maybe there's a great synonym of  `list` that we can use instead, but I can't quite think of one.

Reply:
`from mlflow.deployments import *` is exactly what I am afraid of. I can't find a good synonym too. Unless, no one has a better solution, I'll go with `list` itself then and probably prefer this usage all over the docs
```python
from mlflow import deployments
deployments.list()
```

Reply:
Sorry, should have been clearer - I think it'd actually be better to name them `create_deployment`, `delete_deployment`, `update_deployment`, `list_deployments`, so usage would look like:

```
from mlflow.deployments import *
list_deployments()
```

or

```
from mlflow import deployments
deployments.list_deployments()
```

It makes the method names more verbose, but avoids the conflict with `list` and would also us to be consistent if we eventually promoted deployment functionality to the top-level MLflow fluent API, e.g. via APIs like `mlflow.create_deployment`, `mlflow.update_deployment`, etc.

Reply:
Got it. However, I preferred `xxx` over `xxx_deployment` since that would be the replica of the CLI commands. Plus another major concern I have is about the human language aesthetic. For instance:
`list_deployments` is accurate because it can list more than one deployment but `create_deployments` is wrong because we are not creating more than one deployment (Note the plurality of the command)

Reply:
I think we'd handle the pluralization by just using the appropriate pluralization for the use case (e.g. we'd define a singular `create_deployment` but a pluralized `list_deployments`)

Reply:
Restating our conversation in slack:
> Really?. Not a big fan of that because I think that makes users life miserable. But, that's probably a personal choice. I'll stick with your suggestion